client/client.go

```diff
@@ -0,0 +1,166 @@
+package client
+
+import (
+	"crypto/sha256"
+	"fmt"
+	"net"
+	"os"
+	"path/filepath"
+	"sort"
+
+	rpc "github.com/keybase/go-framed-msgpack-rpc"
+	"github.com/keybase/search/libsearch"
+	sserver1 "github.com/keybase/search/protocol/sserver"
+	"golang.org/x/net/context"
+)
+
+// Client contains all the necessary information for a KBFS Search Client.
+type Client struct {
+	searchCli   sserver1.SearchServerClient   // The client that talks to the RPC Search Server.
+	directory   string                        // The directory where KBFS is mounted.
+	indexer     *libsearch.SecureIndexBuilder // The indexer for the client.
+	pathnameKey [32]byte                      // The key to encrypt and decrypt the pathnames to/from document IDs.
+}
+
+// CreateClient creates a new `Client` instance with the parameters and returns
+// a pointer the the instance.  Returns an error on any failue.
+func CreateClient(ipAddr string, port int, masterSecret []byte, directory string) (*Client, error) {
+	c, err := net.Dial("tcp", fmt.Sprintf("%s:%d", ipAddr, port))
```

Comment:
And we should probably use an `rpc.NewConnectionWithTransport` object instead of using `net` directly, because it will handle the corner case for us, such as what happens when the server isn't running when the client comes up?